staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```diff
@@ -212,10 +219,17 @@ func (o *AutoscaleOptions) Run() error {
 			return fmt.Errorf("cannot autoscale a %v: %v", mapping.GroupVersionKind.Kind, err)
 		}
 
-		hpa := o.createHorizontalPodAutoscaler(info.Name, mapping)
+		// Create HPA objects for both v2 and v1 APIs.
+		// We support both versions to ensure compatibility.
+		hpaV2 := o.createHorizontalPodAutoscalerV2(info.Name, mapping)
+		hpaV1 := o.createHorizontalPodAutoscalerV1(info.Name, mapping)
 
-		if err := o.Recorder.Record(hpa); err != nil {
-			klog.V(4).Infof("error recording current command: %v", err)
+		// Try to record the command with v2 HPA, if it fails, try with v1 HPA.
+		if err := o.Recorder.Record(hpaV2); err != nil {
```

Comment:
You can't do it this way, because in case of a temporary problem you'll record for hpa/v1, but one of the followup calls might succeeded with hpa/v2 making the entire flow not consistent. You need to extract this entire block: https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go#L215-L245 into a separate function, such that in case of any error the entire flow will be either v2 or v1, not a mix of the two.

Reply:
done